File: server/restaurant/src/main/java/hu/elte/alkfejl/restaurant/entity/response/OrderResponse.java
package hu.elte.alkfejl.restaurant.entity.response;

import hu.elte.alkfejl.restaurant.entity.Product;
import hu.elte.alkfejl.restaurant.entity.Status;
import hu.elte.alkfejl.restaurant.entity.User;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.sql.Timestamp;
import java.util.List;

@Data
@EqualsAndHashCode(callSuper = true)
public class OrderResponse extends BaseResponse {

    private Timestamp createDate;

    private String note;

    private User user;

    private Status status;

    private List<OrderProductResponse> orderProducts;

    private Short total;

    @Data
    @EqualsAndHashCode(callSuper = true)
    public static class OrderProductResponse extends BaseResponse {

        private Byte quantity;

        private Product product;
    }
}

File: client/restaurant/src/app/models/responses/order-response.ts
import { BaseModel } from '../base-model';
import { Status } from '../status';
import { Product } from '../product';

export class OrderResponse extends BaseModel {

    public createDate: number;
    public note: string;
    public status: Status;
    public orderProducts: OrderProductResponse[];
    public total: number;

    constructor(object) {
        super(object.id);
        Object.assign(this, object);
    }
}

export class OrderProductResponse extends BaseModel {

    public quantity: number;
    public product: Product;

    constructor(object) {
        super(object.id);
        Object.assign(this, object);
    }
}

